### PR TITLE
filter files for sub-sets

### PR DESCRIPTION
**Pull request by sjanssen2**
Added a notebook to filter full BIOM tables and/or beta dioversity distance matrices down to a set of given sampleIDs, e.g. those that form the 2k sub-set.

### CODE REVIEW

File: ipynb/filterEMP.ipynb [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+   "source": [
+    "%matplotlib inline\n",
+    "\n",
+    "# make all necessary imports\n",
+    "import pandas as pd\n",
+    "import numpy as np\n",
+    "import sys\n",
+    "import seaborn as sn\n",
+    "import matplotlib.pyplot as plt\n",
+    "import os\n",
```
Comment (tkosciol) [inline ipynb/filterEMP.ipynb]: you're not using `sys`, `seaborn`, `plt` or `os` in this notebook. No need to import them

File: ipynb/filterEMP.ipynb [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+    "import biom\n",
+    "from biom.util import biom_open"
```
Comment (tkosciol) [inline ipynb/filterEMP.ipynb]: If you import `biom` there's no need to import `biom.util.biom_open` separately.

File: ipynb/filterEMP.ipynb [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "#we load the mapping file into a Pandas dataframe\n",
+    "metadata = pd.read_csv(file_mapping, sep=\"\\t\", index_col=0, low_memory=False)\n",
+    "\n",
+    "#we chose a (sub-)set of sample IDs which we want to have in our output set. E.g. all sampleIDs that have a np.True_ in the 'subset_2000' column, i.e. those 2000 samples that form the EMP 2000 sub-set \n",
+    "idx = metadata[metadata.subset_2000 == np.True_].index\n",
```
Comment (tkosciol) [inline ipynb/filterEMP.ipynb]: maybe for clarity change `metadata.subset_200` to `metadata['subset_200']`? It's easier to spot

File: ipynb/filterEMP.ipynb [hunk 4]
```diff
@@ -0,0 +1,254 @@
...
+     "text": [
+      "Loading source biom table ... which holds 27398 samples.\n",
+      "Sub-sampling down to 1999 of which 1 are not in the input BIOM table.\n",
+      "Writing resulting biom table to './filtered.biom'\n",
+      "Generating summary.\n"
+     ]
+    }
+   ],
+   "source": [
+    "filterBiomTable(idx, file_biom, \"./filtered.biom\")"
```
Comment (tkosciol) [inline ipynb/filterEMP.ipynb]: change `./filtered.biom` to variable for clarity.

File: ipynb/filterEMP.ipynb [hunk 5]
```diff
@@ -0,0 +1,254 @@
...
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Sub-sampling down to 2000 of which 1 are not in the input matrix.\n",
+      "  processed 0 %\n"
+     ]
+    }
+   ],
+   "source": [
+    "filterBetadivMatrix(idx, file_betaDiv, \"./filtered_betadiv.txt\")"
```
Comment (tkosciol) [inline ipynb/filterEMP.ipynb]: same as above.

File: ipynb/filterEMP.ipynb (hunk 1 above)
Comment (sjanssen2) [inline ipynb/filterEMP.ipynb]: sys is used for some file magic, but the rest is really not used (any more). I will remove those imports.

File: ipynb/filterEMP.ipynb (hunk 2 above)
Comment (sjanssen2) [inline ipynb/filterEMP.ipynb]: Your right

File: ipynb/filterEMP.ipynb (hunk 3 above)
Comment (sjanssen2) [inline ipynb/filterEMP.ipynb]: done.

File: ipynb/filterEMP.ipynb (hunk 4 above)
Comment (sjanssen2) [inline ipynb/filterEMP.ipynb]: done and restructured the cell for setting those variables a bit

File: ipynb/filterEMP.ipynb (hunk 5 above)
Comment (sjanssen2) [inline ipynb/filterEMP.ipynb]: done.

**Comment by josenavas** [comment]
@sjanssen2 @tkosciol I highly recommend to take a look at QIIME's scripts list [here](http://qiime.org/scripts/) and get familiar with it.

Scripts to filter biom tables and/or distance matrices based on a list of sample ids are already written (see [here](http://qiime.org/scripts/filter_samples_from_otu_table.html) and [here](http://qiime.org/scripts/filter_distance_matrix.html) ).

There are a lot of utility script (that can be called from an ipython notebook using the `!` notation) that already exists and there is no need to re-invent the wheel. Investing some time on learning Qiime will highly reduce to overall developing time.

In case of doubt, drop a quick line to any of us since we probably already know the script that does or facilitates most of the work that needs to be done in the EMP. Note that the difficulty of the EMP is running the stuff rather than code development.

Another advantage is that all QIIME code is unit-tested, while none of the code being added here in the form of notebooks is not unit-tested.

**Comment by sjanssen2** [comment]
Hi Jose.
Good points. But I created this notebook mainly to ease the filtering process for EMP users that are not too familiar with qiime coding. 

**The BIOM filtering** 
is basicall a wrapper around: biom.load_table biom.filter and biom.util.biom_open --> to_hdf5 Thus, I would not see this as a re-invention. It's more an elaborate documentation. 

**distance filtering**
Unfortunately, filter_distance_matrix.py is quite slow. It takes ~6min to process my input file. I could speed this up 6-fold. But you are right, no unit tests :-/

**Comment by josenavas** [comment]
Following-up over email
